src/Host/Client/Impl/Definitions/IRExpressionEvaluator.cs

```diff
@@ -13,62 +14,146 @@ namespace Microsoft.R.Host.Client {
     [Flags]
     public enum REvaluationKind {
         Normal = 0,
+        /// <summary>
+        /// Allows other evaluations to occur while this evaluation is ongoing.
+        /// </summary>
         Reentrant = 1 << 1,
+        /// <summary>
+        /// Indicates that the result of this evaluation is a value that should be serialized to JSON.
+        /// When this flag is set, <see cref="REvaluationResult.StringResult"/> will be null, but
+        /// <see cref="REvaluationResult.JsonResult"/> will contain the value after it has been deserialized.
+        /// </summary>
         Json = 1 << 2,
+        /// <summary>
+        /// Indicates that this expression should be evaluated in the base environment (<c>baseenv()</c>).
+        /// Not compatible with <see cref="EmptyEnv"/>.
+        /// </summary>
         BaseEnv = 1 << 3,
+        /// <summary>
+        /// Indicates that this expression should be evaluated in the empty environment (<c>emptyenv()</c>).
+        /// Not compatible with <see cref="BaseEnv"/>.
+        /// </summary>
         EmptyEnv = 1 << 4,
+        /// <summary>
+        /// Allows this expression to be cancelled.
+        /// </summary>
+        /// <remarks>
+        /// If this evaluation happens as a nested evaluation inside some other <seealso cref="Reentrant"/>
+        /// evaluation, it must be <seealso cref="Cancelable"/> for the outer evaluation to be cancelable.
+        /// </remarks>
         Cancelable = 1 << 5,
+        /// <summary>
+        /// Indicates that this expression should be evaluated in a fresh blank environment, that has the
+        /// otherwise designated target environment set as its parent, and which is discarded immediately
+        /// after evaluation.
+        /// </summary>
+        /// <remarks>
+        /// Similar to <c>local()</c>, in that it will contain and discard any assignments to local variables
+        /// that are performed as part of evaluation.
```

Comment:
Actually, on a second thought, I think I'll get rid of this altogether. It doesn't do anything that `local()` won't do, and we don't use it anywhere - it's really just a leftover from back when I tried to use this as a workaround for the issue when evals were breaking into debugger because of debug bits (which has long since been fixed properly). I won't touch the host for now, but get rid of this here so that no-one tries to use it.